dpnp/backend/custom_kernels.cpp

```diff
@@ -161,6 +163,60 @@ void custom_blas_dot_c(void* array1_in, void* array2_in, void* result1, size_t s
 template void custom_blas_dot_c<long>(void* array1_in, void* array2_in, void* result1, size_t size);
 template void custom_blas_dot_c<int>(void* array1_in, void* array2_in, void* result1, size_t size);
 
+template <typename _DataType>
+void custom_lapack_syevd_c(void* array_in, void* result1, size_t size)
+{
+    if (!size)
+    {
+        return;
+    }
+
+    _DataType* array = reinterpret_cast<_DataType*>(array_in);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
+
+    if constexpr (std::is_same<_DataType, double>::value || std::is_same<_DataType, float>::value)
```

Comment:
this code works for `long` and `int`. No reason to limit it by floating point types.